src/mongo_types.ts

```diff
@@ -442,6 +445,28 @@ export class TypedEventEmitter<Events extends EventsDescription> extends EventEm
       this.mongoLogger?.debug(this.component, loggableHeartbeatEvent);
     }
   }
+  /** @internal */
+  emitAndLogCommand<EventKey extends keyof Events>(
+    monitorCommands: boolean,
+    event: EventKey | symbol,
+    databaseName: string,
+    hello: Document | null,
+    ...args: Parameters<Events[EventKey]>
+  ): void {
+    if (monitorCommands) {
```

Comment:
If monitorCommands is true, and hello is falsey, then wouldn't that mean we would emit command started events for handshakes?

Reply:
Yes. We only want to not log hello commands, emitting is fine. See [here](https://github.com/mongodb/specifications/blob/master/source/command-logging-and-monitoring/command-logging-and-monitoring.rst#log-messages).

Reply:
Is this condition still necessary with the "established" condition & fix?

Reply:
Ah ok, because this function can be entered if monitoring OR logging is enabled, we need to filter inside here. fair enough.